### PR TITLE
lightning: add back table empty check and add a switch config

### LINKED ISSUE #27919: lightning: lightning doesn't check whether target table is empty  (behavior changed compared to 5.0.x)
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Lightning 5.0.1 added a check to ensure target table is empty before import for DBaaS.
Lightning 5.1.0 removed this check due to new feature parallel import / incremental import, this changes previous product behaviors.

Steps:
1. Import a table to TiDB with lightning
2. Import the table again to TiDB again with lightning

### 2. What did you expect to see? (Required)
Lightning behavior should be consistent for 5.0.1 and 5.1.0+.

### 3. What did you see instead (Required)
For lightning 5.1.0+, lightning doesn't check target table empty or not before the import, this may result in:
- Duplicate rows inserted into the table, if the import data doesn't have PK/UK
- Import failed with checksum mismatched error, if import data has PK/UK

### 4. What is your TiDB version? (Required)
5.2.0

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: close #27919

Problem Summary:

### What is changed and how it works?
- Add a new config `tikv-importer.incremental-import` to control whether allow lightning to import to table that already contains data
- When tikv-importer.incremental-import = false, tidb-lightning will check target tables' vacancy. If the tables already contains data, lightning will directly report an error and exit. In this mode, lightning will also avoid create and write data into  `lightning_meta` in target tidb.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility
  - Add a new config to control the `table empty check`, so this is not compatible with v5.2 and v5.3.

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Add a new config to control whether support incremental import.
```

### CODE REVIEW

**Comment by glorv** [comment]
@sunzhaoyang Please also take a look.

File: br/pkg/lightning/restore/meta_manager.go [hunk 1]
```diff
@@ -1027,9 +1027,65 @@ func (m noopTableMetaMgr) UpdateTableBaseChecksum(ctx context.Context, checksum
...
+}
+
+func (b singleMgrBuilder) TaskMetaMgr(pd *pdutil.PdController) taskMetaMgr {
+	return &singleTaskMetaMgr{
+		pd: pd,
+	}
+}
+
+func (b singleMgrBuilder) TableMetaMgr(tr *TableRestore) tableMetaMgr {
+	return noopTableMetaMgr{}
```
Comment (3pointer) [inline br/pkg/lightning/restore/meta_manager.go:1050]: 🤔

Comment (glorv) [reply br/pkg/lightning/restore/meta_manager.go:1050]: For single tidb-lightning, The table meta data is useless. So we just do nothing here.

File: br/pkg/lightning/restore/check_info.go [hunk 2]
```diff
@@ -1053,3 +1057,86 @@ outloop:
...
+			return nil
+		}
+		return errors.Trace(err)
+	}
+
+	if len(tableNames) > 0 {
+		// sort the failed names
+		sort.Slice(tableNames, func(i, j int) bool {
+			return strings.Compare(tableNames[i], tableNames[j]) < 0
+		})
```
Comment (kennytm) [inline br/pkg/lightning/restore/check_info.go:1122]:
```suggestion
		sort.Strings(tableNames)
```

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Review by kennytm — APPROVED** [review]

**Pull request merged**